src/game/client/draw/w3dtanktruckdraw.cpp

```diff
@@ -0,0 +1,651 @@
+#include "mapper.h"
+#include "matinfo.h"
+#include "object.h"
+#include "particlesys.h"
+#include "particlesysmanager.h"
+#include "physicsupdate.h"
+#include "scriptengine.h"
+#include "view.h"
+
+W3DTankTruckDrawModuleData::W3DTankTruckDrawModuleData() :
+    m_treadDebrisNameLeft("TrackDebrisDirtLeft"),
+    m_treadDebrisNameRight("TrackDebrisDirtRight"),
+    m_treadAnimationRate(0.0f),
+    m_treadPivotSpeedFraction(0.6f),
+    m_treadDriveSpeedFraction(0.3f)
+{
+}
+
+void W3DTankTruckDrawModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    // clang-format off
+    static FieldParse dataFieldParse[] = {
+        { "Dust", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_dustEffectName) },
+        { "DirtSpray", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_dirtEffectName) },
+        { "PowerslideSpray", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_powerslideEffectName) },
+        { "LeftFrontTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_frontLeftTireBoneName) },
+        { "RightFrontTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_frontRightTireBoneName) },
+        { "LeftRearTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_rearLeftTireBoneName) },
+        { "RightRearTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_rearRightTireBoneName) },
+        { "MidLeftFrontTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_midFrontLeftTireBoneName) },
+        { "MidRightFrontTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_midFrontRightTireBoneName) },
+        { "MidLeftRearTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_midRearLeftTireBoneName) },
+        { "MidRightRearTireBone", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_midRearRightTireBoneName) },
+        { "TireRotationMultiplier", &INI::Parse_Real, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_rotationSpeedMultiplier) },
+        { "PowerslideRotationAddition", &INI::Parse_Real, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_powerslideRotationAddition) },
+        { "TreadDebrisLeft", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_treadDebrisNameLeft) },
+        { "TreadDebrisRight", &INI::Parse_AsciiString, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_treadDebrisNameRight) },
+        { "TreadAnimationRate", &INI::Parse_Velocity_Real, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_treadAnimationRate) },
+        { "TreadPivotSpeedFraction", &INI::Parse_Real, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_treadPivotSpeedFraction) },
+        { "TreadDriveSpeedFraction", &INI::Parse_Real, nullptr, offsetof(W3DTankTruckDrawModuleData,  m_treadDriveSpeedFraction) },

```

Comment:
Style: there are 2 white spaces within the `offsetof` call in each line.

Reply:
fixed,